### ISSUE TITLE
[Bug] FA4 Blackwell hd256 forward skips Q/K contiguous conversion on compiled-kernel cache hits

### ISSUE THREAD

**Issue by mabingqi** [Dao-AILab/flash-attention#2886]
# [Bug] FA4 Blackwell hd256 forward skips contiguous conversion on kernel-cache hits

On Blackwell, the dedicated `head_dim=256` forward kernel assumes contiguous Q/K/V layouts. In `flash_attn/cute/interface.py`, the `.contiguous()` calls are inside the `compile_key not in _flash_attn_fwd.compile_cache` branch. Cache hits skip this conversion and produce incorrect results for non-contiguous Q/K, even when `stride(-1) == 1`.

This is the **compiled GPU kernel cache**.

## Environment

GB300 (SM 10.3), PyTorch `2.12.1+cu130`, FA4 `4.0.0b19.dev2+g940cd96`, CUTLASS DSL `4.6.0.dev0`.
Tested checkout: `940cd9680f3315f2f06b43ab5bea2c2cf2d96806` with pre-existing local changes; pristine upstream not separately tested.

## Minimal reproduction

Run in a fresh process. No model weights are needed.

```python
import torch
from flash_attn.cute import flash_attn_func

torch.manual_seed(0)
kw = dict(device="cuda", dtype=torch.bfloat16)
q = torch.randn(1, 24, 256, 256, **kw).transpose(1, 2)
k = torch.randn(1, 4, 256, 256, **kw).transpose(1, 2)
v = torch.randn(1, 256, 4, 256, **kw)
assert not q.is_contiguous() and not k.is_contiguous()
assert q.stride(-1) == k.stride(-1) == 1

def attention(q, k):
    out = flash_attn_func(q, k, v, causal=True)
    return out[0] if isinstance(out, tuple) else out

with torch.inference_mode():
    expected = attention(q.contiguous(), k.contiguous())  # Populate kernel cache.
    actual = attention(q, k)  # Same values and shapes, different strides.
    print("max_abs_error:", (actual - expected).abs().max().item())
    torch.testing.assert_close(actual, expected, rtol=1e-2, atol=1e-2)
```

Observed: `max_abs_error: 4.5625`; the assertion fails with **94.0% mismatched elements**.
Expected: a layout-only change should preserve the attention result.

## Minimal fix

Move the existing contiguous conversion outside the cache-miss branch, before the cache key and Cute tensor wrappers are constructed:

```diff
--- a/flash_attn/cute/interface.py
+++ b/flash_attn/cute/interface.py
@@
-    compile_key = (
+    if use_dedicated_hd256_kernel and qv is None:
+        q, k, v = q.contiguous(), k.contiguous(), v.contiguous()
+
+    compile_key = (
@@
-                    q = q.contiguous() if not q.is_contiguous() else q
-                    k = k.contiguous() if not k.is_contiguous() else k
-                    v = v.contiguous() if not v.is_contiguous() else v
```

Remove the old conversion comment with the moved block. This ensures layout normalization runs on every invocation, including cache hits.

Verified with the same reproducer and a process-local application of this change: maximum absolute error drops from **4.5625 to 0.0**, and the assertion passes.
